### PR TITLE
[Mvc] Adds a tag helper for rendering components

### PR DESCRIPTION
**Pull request by javiercn**
* Adds a tag helper to render components

### CODE REVIEW

File: src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
+﻿using System;
```
Comment (rynowak) [inline src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs]: needs copyright

File: src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Mvc.Rendering;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Microsoft.AspNetCore.Razor.TagHelpers;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.TagHelpers
+{
+    [HtmlTargetElement("component", Attributes = ComponentTypeName)]
```
Comment (rynowak) [inline src/Mvc/Mvc.TagHelpers/src/ComponentTagHelper.cs]: Needs docs

File: src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs [hunk 3]
```diff
@@ -67,6 +67,38 @@ public static class HtmlHelperComponentExtensions
...
+            this IHtmlHelper htmlHelper,
+            Type type,
+            RenderMode renderMode,
+            object parameters)
+        {
+            if (htmlHelper == null)
+            {
+                throw new ArgumentNullException(nameof(htmlHelper));
+            }
+
```
Comment (rynowak) [inline src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs]: We should have a null check here for type

Comment (rynowak) [reply src/Mvc/Mvc.ViewFeatures/src/HtmlHelperComponentExtensions.cs]: What's the behaviour here if you pass in a non-component type. Does it blow up with a sensible error or should we add a check?

File: src/Mvc/samples/MvcSandbox/Pages/Components.cshtml [hunk 4]
```diff
@@ -15,8 +16,13 @@
     <link href="css/site.css" rel="stylesheet" />
 </head>
 <body>
-    <app>@(await Html.RenderComponentAsync<MvcSandbox.Components.App>(RenderMode.Static))</app>
+    <app>
+        <component 
+          component-type="typeof(MvcSandbox.Components.App)"
+          render-mode="ServerPrerendered"
+          parameter-Name="@("Daniel")"></component>
```
Comment (rynowak) [inline src/Mvc/samples/MvcSandbox/Pages/Components.cshtml]: Does this work? Or is this assuming your other PR

**Comment by pranavkm** [comment]
Closing this in favor of https://github.com/aspnet/AspNetCore/pull/14592/files

**Pull request closed without merging**